packages/swap/src/adapters/api/dexhunter/transformers.test.ts

```diff
@@ -0,0 +1,133 @@
+
+import {
+  fromSwapProtocol,
+  ptIdDh,
+  toSwapProtocol,
+  transformersMaker,
+} from './transformers'
+import {api, primaryTokenInfo} from './api.mocks'
+import {Dex} from './types'
+
+const address =
+  'addr1q9qhyvkm5fytm5ckgshny0zz08a3urhhh7ckdqxcm27av40eafn3v5lr2w2n2er9uj7c743mt42gpe8tgek6394z9t7qn4yjzl'
+const network = Chain.Network.Mainnet
+const transformers = transformersMaker({
+  primaryTokenInfo,
+  address,
+  network,
+  isPrimaryToken,
+})
+
+describe('transformers', () => {
+  describe('tokens', () => {
+    test('should correctly transform the tokens response', () => {
+      expect(transformers.tokens.response(api.responses.tokens)).toEqual(
+        api.results.tokens,
+      )
+    })
+
+    test('should correctly transform the token id', () => {
+      expect(transformers.tokens.fromId(ptIdDh)).toBe(primaryTokenInfo.id)
+      expect(transformers.tokens.toId(primaryTokenInfo.id)).toBe('ADA')
+    })
+  })
+
+  describe('orders', () => {
+    test('should correctly transform the orders response', () => {
+      expect(transformers.orders.response(api.responses.orders)).toEqual(
+        api.results.orders,
+      )
+    })
+  })
+
+  describe('cancel', () => {
+    test('should correctly transform the cancel request', () => {
+      expect(transformers.cancel.request(api.inputs.cancel)).toEqual(
+        api.requests.cancel(address),
+      )
+    })
+
+    test('should correctly transform the cancel response', () => {
+      expect(transformers.cancel.response(api.responses.cancel)).toEqual(
+        api.results.cancel,
+      )
+    })
+  })
+
+  describe('estimate', () => {
+    test('should correctly transform the estimate request', () => {
+      expect(transformers.estimate.request(api.inputs.estimate)).toEqual(
+        api.requests.estimate,
+      )
+    })
+
+    test('should correctly transform the estimate response', () => {
+      expect(transformers.estimate.response(api.responses.estimate)).toEqual(
+        api.results.estimate,
+      )
+    })
+  })
+})
+
+describe('toSwapProtocol', () => {
+  it.each`
+    dex                   | protocol
+    ${Dex.Minswap_v1}     | ${Swap.Protocol.Minswap_v1}
+    ${Dex.Minswap_v2}     | ${Swap.Protocol.Minswap_v2}
+    ${Dex.Wingriders_v1}  | ${Swap.Protocol.Wingriders_v1}
+    ${Dex.Wingriders_v2}  | ${Swap.Protocol.Wingriders_v2}
+    ${Dex.Vyfi_v1}        | ${Swap.Protocol.Vyfi_v1}
+    ${Dex.Sundaeswap_v1}  | ${Swap.Protocol.Sundaeswap_v1}
+    ${Dex.Sundaeswap_v3}  | ${Swap.Protocol.Sundaeswap_v3}
+    ${Dex.Splash_v1}      | ${Swap.Protocol.Splash_v1}
+    ${Dex.Muesliswap_v2}  | ${Swap.Protocol.Muesliswap_v2}
+    ${Dex.Muesliswap_clp} | ${Swap.Protocol.Muesliswap_clp}
+  `('should map $dex to $protocol', ({dex, protocol}) => {
+    expect(toSwapProtocol(dex)).toBe(protocol)
+  })
+})
+
+describe('fromSwapProtocol', () => {
+  // TODO: @jorbuedo check if the undefineds are correct, cuz the API returns when is not from dexhunter
```

Comment:
@jorbuedo there is a different behaviour when falling back for DH x Muesli, can you confirm it? (DH -> Splash, but not within the transformer, when Muesli does fallback in the transformer to Muesli)

Reply:
A few of the defaults are just there because the types need something, but should not happen if everything goes correctly. But can't really tell if everything is going correctly.
I don't have a good sample of requests and responses to cover all, so it's just guesswork from what I saw in their app, which also changes from time to time.

Reply:
Could be an option yes, and just not let swaps happen for unknown dex. Less troublesome than defaulting to a wrong one.